mantidproject/mantid
# ILL :: Indirect Reduction Framework

This PR implements the reduction framework for IN16B indirect geometry instrument at ILL.
It includes a new version for the loader (which treats single detectors slightly different), 3 new workflow algorithms (deprecating the existing 2) and corresponding upgrade of the GUI. 

The results of the reduction are preliminarily validated against the existing software at ILL (LAMP), and science validation is being run in parallel.

It turned out to be a rather large ticket, but the different parts are in tight relation and have been developed together, so we were not able to break it down to smaller pieces.

**To test:**


1. Use this example data (see the READMEs inside)
[FWS.zip](https://github.com/mantidproject/mantid/files/654308/FWS.zip)
[QENS-1wing.zip](https://github.com/mantidproject/mantid/files/654306/QENS-1wing.zip)
[QENS-2wing.zip](https://github.com/mantidproject/mantid/files/654307/QENS-2wing.zip)
2. Read the documentations.
2. Set ILL/IN16B as facility/instrument.
3. Go to Interfaces -> Indirect -> Data Reduction -> Energy Transfer tab.
4. Run with different options/combinations. Try also to break it.
5. Code review.

Fixes #16753 .
Fixes #17559 .
Fixes #16589 .
Fixes #14179 .
Fixes #14649 .
Fixes #14650 .
Fixes #14653 .

Some of these issues were opened historically, and are no longer relevant.



Release notes have been updated.

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
